pkg/rtc/participant.go

```diff
@@ -666,50 +668,84 @@ func (p *ParticipantImpl) GetAudioLevel() (level uint8, active bool) {
 	return
 }
 
-func (p *ParticipantImpl) GetConnectionQuality() livekit.ConnectionQuality {
+func (p *ParticipantImpl) GetConnectionQuality() *livekit.ConnectionQualityInfo {
 	// avg loss across all tracks, weigh published the same as subscribed
 	var pubLoss, subLoss uint32
 	var reducedQualityPub bool
 	var reducedQualitySub bool
+	var audioScores float64
+	var numAudioTracks int
 	p.lock.RLock()
 	defer p.lock.RUnlock()
 	for _, pubTrack := range p.publishedTracks {
 		if pubTrack.IsMuted() {
 			continue
 		}
-		pubLoss += pubTrack.PublishLossPercentage()
-		publishing, registered := pubTrack.NumUpTracks()
-		if registered > 0 && publishing != registered {
-			reducedQualityPub = true
+		// audio has scores available
+		if pubTrack.Kind() == livekit.TrackType_AUDIO {
+			audioScores += pubTrack.GetConnectionScore()
+			numAudioTracks++
+		} else {
+			pubLoss += pubTrack.PublishLossPercentage()
+			publishing, registered := pubTrack.NumUpTracks()
+			if registered > 0 && publishing != registered {
+				reducedQualityPub = true
+			}
 		}
 	}
-	numTracks := uint32(len(p.publishedTracks))
-	if numTracks > 0 {
-		pubLoss /= numTracks
-	}
 
 	for _, subTrack := range p.subscribedTracks {
 		if subTrack.IsMuted() {
 			continue
 		}
-		if subTrack.DownTrack().GetForwardingStatus() != sfu.ForwardingStatusOptimal {
-			reducedQualitySub = true
+		if subTrack.DownTrack().Kind() == webrtc.RTPCodecTypeAudio {
+			audioScores += subTrack.DownTrack().GetConnectionScore()
```

Comment:
similar suggestion for `DownTrack.GetConnectionScore`, can we make the interface consistent for both video/audio?